File: Days.32/C++/2.Postfix-to-Prefix.cpp
// Postfix ----> Prefix
#include<iostream>
#include<vector>
using namespace std;

string PostfixtoPrefix(string &str)
{
    vector<string> op;
    string a,b,temp;
    for(int i=0; i<str.size(); i++)
    {
        if (str[i]=='+' or str[i]=='-' or str[i]=='*' or str[i]=='/' or str[i]=='^')
        {
            b=op.back();
            op.pop_back();
            a=op.back();
            op.pop_back();
            op.push_back(str[i]+a+b);
        }
        else
        {
            temp=str[i];
            op.push_back(temp);
        }
    }
    return op.back();
}

int main()
{
    string str;
    cin>>str;
    string ans=PostfixtoPrefix(str);
    cout<<ans<<"\n";
    return 0;
}

File: Days.32/Python/2.Postfix-to-Prefix.py
# Postfix ----> Prefix


def PostfixtoPrefix(str):
    op=[]
    for i in range(len(str)):
        if str[i]=='+' or str[i]=='-' or str[i]=='*' or str[i]=='/' or str[i]=='^':
            b=op.pop()
            a=op.pop()
            op.append(str[i]+a+b)
        else:
            op.append(str[i])
    return op[-1]
            

def main():
    str=input()
    ans=PostfixtoPrefix(str)
    print(ans)


main()

